src/models/userModel.js

```diff
@@ -0,0 +1,47 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+
+const Schema = mongoose.Schema;
+
+let userSchema = new Schema({
+  email: {
+    type: String,
+    required: true,
+    unique: true,
+  },
+  password: {
+    type: String,
+    required: true,
+  },
+  firstName: {
+    type: String,
+    required: true,
+  },
+  lastName: {
+    type: String,
+    required: true,
+  },
+  role: {
+    type: String,
+    default: "ROLE_USER",
+  },
+});
+
+// Pre save to hash password
+userSchema.pre("save", function preSave(next) {
```

Comment:
Useless named function, but it's not a big deal so do what you want
```suggestion
userSchema.pre("save", function (next) {
```